modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQConsumer.java

```diff
@@ -34,255 +34,220 @@
-			this.message = message;
-			return this;
-		}
+    /**
+     * Set ID of this RabbitMQ consumer
+     *
+     * @param id ID
+     */
+    @Override
+    public void setId(int id) {
+        idString = "[" + store.getName() + "-C-" + id + "]";
+    }
 
-		public boolean ack() {
-			if (message != null && channel != null && channel.isOpen()) {
-				try {
-					channel.basicAck(message.getEnvelope().getDeliveryTag(), false);
-					return true;
-				} catch (IOException e) {
-					logger.error(getId() + " cannot ack last read message. Error:"
-					             + e.getLocalizedMessage(), e);
-					return false;
-				}
-			}
-			return false;
-		}
+    /**
+     * Set the {@link Connection} object
+     *
+     * @param connection a {@link Connection} object
+     */
+    public void setConnection(Connection connection) {
+        this.connection = connection;
+    }
 
-		public GetResponse getMessage() {
-			return message;
-		}
+    /**
+     * Set the {@link Channel} object
+     *
+     * @param channel a {@link Channel} object
+     */
+    public void setChannel(Channel channel) {
+        this.channel = channel;
+    }
 
-		public CachedMessage setMc(MessageContext mc) {
-			this.mc = mc;
-			return this;
-		}
+    /**
+     * Set the queue name
+     *
+     * @param queueName the queue name
+     */
+    public void setQueueName(String queueName) {
+        this.queueName = queueName;
+    }
 
-		public CachedMessage setId(String id) {
-			this.id = id;
-			return this;
-		}
+    /**
+     * Update the cached message with the message received from the queue
+     *
+     * @param delivery  the received message from the queue
+     * @param messageId the message id
+     */
+    private void updateCache(GetResponse delivery, String messageId) {
+        cachedMessage.setMessage(delivery);
+        cachedMessage.setId(messageId);
+    }
 
-		public String getId() {
-			return id;
-		}
-	}
+    /**
+     * This is used to store the last received message
+     * if the message is successfully sent to the endpoint, ack will sent to the store
+     * in order to delete message from the queue
+     * <p/>
+     * In RabbitMQ message ack should be using the same channel which was consumed the message
+     * There for the consumed channel will also stored without closing until the message is ackd
+     */
+    private final class CachedMessage {
+        private GetResponse message = null;
+        private String id = "";
+
+        public void setMessage(GetResponse message) {
+            this.message = message;
+        }
+
+        public boolean ack() {
+            if (message != null && channel != null && channel.isOpen()) {
+                try {
+                    channel.basicAck(message.getEnvelope().getDeliveryTag(), false);
+                    return true;
+                } catch (IOException e) {
+                    log.error(getId() + " cannot ack last read message. Error: " + e.getLocalizedMessage(), e);
```

Comment:
We cannot ack using the channel. Maybe it is broken. Should we clean it up and reset? As I saw, if we could not ack,  we just retry the same message right?

Reply:
As per the implementation, the processor should be deactivated in case of a connection error with the broker. The processor implementation handled it based on the boolean value returned by the consumer. The `ack` method in the RabbitMQConsumer invoked by the processor upon successful backend invocation. The method must return whether it is successful or not.